### DISCUSSION TITLE
Implement new Contract Design

### DISCUSSION THREAD

**Discussion by nielstron** [OpShin/opshin#557] [Ideas / Improvements]
The contract design proposed in #306 should be implement with the following additions

- `opshin build` recursively searches the current directory for python files containing contracts. all found contracts are built.
- the current functionality is mainted in a form similar to `aiken export` (https://github.com/aiken-lang/aiken/pull/843) to be reused for i.e. [Plutus Bench](https://projectcatalyst.io/funds/11/cardano-open-developers/plutus-bench-test-measure-inspect-smart-contracts)

> [Bug bounty](https://github.com/OpShin/opshin/blob/main/BUG_BOUNTY.md): 1500 ADA

**Comment by irispigo** [comment]
This issue wasn't solved, yet? If so, I'll fix it.

**Comment by nielstron** [comment]
No, not fixed yet :)

**Comment by nielstron** [comment]
Closing as duplicate of #306
